### PR TITLE
DOCSP-41141: Indexes

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-kotlin-sync/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-41141>
Staging - <https://preview-mongodbmcmorisi.gatsbyjs.io/kotlin-sync/DOCSP-41141-indexes/indexes/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/indexes.txt [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+..    :titlesonly:
+..    :maxdepth: 1
+
+..    /work-with-indexes
+
+Overview
+--------
+
+On this page, you can see copyable code examples that show how to work with common
+types of indexes that you can use with the {+driver-short+}.
```
Comment (rustagir) [inline source/indexes.txt:32]:
```suggestion
On this page, you can see copyable code examples that show how to manage different
types of indexes by using the {+driver-short+}.
```

File: source/indexes.txt [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+   :end-before: end-multikey
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+.. TODO To learn more about multikey indexes, see the :ref:`kotlin-sync-multikey-index`
+.. guide.
+
+Atlas Search Index
+------------------
```
Comment (rustagir) [inline source/indexes.txt:103]: S: This section could be called Atlas Search Index Management

File: source/indexes.txt [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
```
Comment (rustagir) [inline source/indexes.txt:104]: S: add a quick intro sentence
"The following sections describe how to manage Atlas Search indexes.."

File: source/indexes.txt (hunk 2 above)
Comment (rustagir) [inline source/indexes.txt:103]: Q: what is the rationale to have this section in this location in the TOC?

File: source/indexes.txt [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+   :copyable:
+   :dedent:
+
+.. TODO: To learn more about listing search indexes, see the :ref:`kotlin-sync-atlas-search-index-list`
+.. guide.
+
+Update Search Indexes
+~~~~~~~~~~~~~~~~~~~~~
+
+The following example updates an existing Atlas search index:
```
Comment (rustagir) [inline source/indexes.txt:141]: S: describe the update
```suggestion
The following example updates an existing Atlas search index to ...:
```

File: source/indexes.txt [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+   :copyable:
+   :dedent:
+
+.. TODO: To learn more about deleting search indexes, see the :ref:`kotlin-sync-atlas-search-index-drop`
+.. guide.
+
+Text Index
+----------
+
+The following example creates a text index on the specified field:
```
Comment (rustagir) [inline source/indexes.txt:171]:
```suggestion
The following example creates a text index on the specified string field:
```

File: source/indexes.txt [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+   :start-after: start-clustered
+   :end-before: end-clustered
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+.. TODO: To learn more about wildcard indexes, see the :ref:`kotlin-sync-clustered-index`
+.. guide.
+
+Remove an Index
```
Comment (rustagir) [inline source/indexes.txt:245]: S: to match search indexes heading
```suggestion
Delete an Index
```

**Review by rustagir — CHANGES_REQUESTED** [review]
a few small things, but I'd like to take another look!

File: source/indexes.txt (hunk 2 above)
Comment (mcmorisi) [reply source/indexes.txt:103]: The honest answer is that I just grabbed the format from Pymongo's equivalent. I reached out to Jordan and he mentioned his placement was based on the [Java](https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/indexes/) page.

Do you have strong feelings about shifting the section up or down?

Comment (rustagir) [reply source/indexes.txt:103]: I dont have strong feelings, but i would suggest moving the section to the end of the list of index types as this is a separate index category with different methods

**Review by rustagir — APPROVED** [review]
lgtm!

**Pull request merged**